pydm/widgets/drawing.py

```diff
@@ -1184,3 +1185,34 @@ def resetPoints(self):
         self.update()
 
     points = Property("QStringList", getPoints, setPoints, resetPoints)
+
+
+class PyDMDrawingIrregularPolygon(PyDMDrawingPolyline):
+    """
+    A widget contains an irregular polygon (arbitrary number of vertices, arbitrary lengths).
+
+    This is a special case of the PyDMDrawingIrregularPolygon, adding the requirement that 
+    the last point is always identical to the first point.
+
+    This widget is created for compatibility with MEDM's *polygon* widget.
+
+    Parameters
+    ----------
+    parent : QWidget
+        The parent widget for the Label
+    init_channel : str, optional
+        The channel to be used by the widget.
+    """
+
+    def setPoints(self, points):
+        if points[0] != points[-1]:
+            points.append(points[0])  # close the polygon
```

Comment:
Since this is done prior to the validator, we can have a case where points are given as a string from designer and have the same "value" in terms of coordinates while not matching here, creating a duplicate point.

Reply:
Needs a logic change.  Seems OK to run through the validator first so the comparison is on same terms.